correct syntax for the controller callbacks

papayaah opened the pull request:
@bryanmtl an issue has been reported that cancellation/pause email notifications have stopped. the reason is the syntax were incorrect.

cc @hugobast @braidn

papayaah committed:
correct syntax for the controller callbacks

bryanmtl commented:
@mickeyren This looks fine to me. Is it normal that the build is failing, though?

papayaah committed:
fix travis script
ensure a working versionr Rakefile

papayaah commented:
i updated the travis command, we can now just wait and see if everything will be green.

papayaah commented:
there's a single error @bryanmtl . is this still valid? ```Stripe.api_key = "sk_test_jTiNI1BxjFxBr4TUqdHefc1f"```

braidn commented:
@mickeyren this also looks fine. What happens if you add a random, test api key? Does it pass?

papayaah commented:
I actually executed the suite locally and everything just went fine:

```
Finished in 46.17 seconds (files took 3.99 seconds to load)
81 examples, 0 failures, 1 pending
```

bryanmtl commented:
Has that key been added to Travis?

papayaah commented:
@bryanmtl it's actually hard coded

bryanmtl commented:
The tests pass locally but are failing on Travis?

papayaah commented:
yeah it passes locally.

hugobast commented:
I'm giving this a hot minute or 20.

hugobast commented:
Well 20 minutes have passed and after suffering the install ruby 2.2.3, bundler won't bundle this project...

papayaah commented:
hmm... some gem it can't install?

hugobast commented:
Right; I think solidus master has moved too far ahead. Libraries like this don't have a Gemfile.lock so we can't ensure what version we end up with a month, six or a year later. I've pulled the sha from Travis. Now I'm fighting `ruby_dep` that's yelling I need to update to 2.2.5+.

hugobast commented:
Working through this new problem now; post a solution in the meantime if you guys have one:

```
▶ bundle exec rake --trace
rake aborted!
NoMethodError: undefined method `last_comment' for #<Rake::Application:0x007fb59696ea80>
/Users/hugobastien/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/rspec-core-3.3.2/lib/rspec/core/rake_task.rb:82:in `define'
/Users/hugobastien/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/rspec-core-3.3.2/lib/rspec/core/rake_task.rb:62:in `initialize'
/Users/hugobastien/src/solidus_subscriptions/Rakefile:7:in `new'
/Users/hugobastien/src/solidus_subscriptions/Rakefile:7:in `<top (required)>'
/Users/hugobastien/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/rake-12.0.0/lib/rake/rake_module.rb:28:in `load'
/Users/hugobastien/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/rake-12.0.0/lib/rake/rake_module.rb:28:in `load_rakefile'
/Users/hugobastien/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/rake-12.0.0/lib/rake/application.rb:687:in `raw_load_rakefile'
/Users/hugobastien/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/rake-12.0.0/lib/rake/application.rb:96:in `block in load_rakefile'
/Users/hugobastien/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/rake-12.0.0/lib/rake/application.rb:178:in `standard_exception_handling'
/Users/hugobastien/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/rake-12.0.0/lib/rake/application.rb:95:in `load_rakefile'
/Users/hugobastien/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/rake-12.0.0/lib/rake/application.rb:79:in `block in run'
/Users/hugobastien/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/rake-12.0.0/lib/rake/application.rb:178:in `standard_exception_handling'
/Users/hugobastien/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/rake-12.0.0/lib/rake/application.rb:77:in `run'
/Users/hugobastien/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/rake-12.0.0/exe/rake:27:in `<top (required)>'
/Users/hugobastien/.rbenv/versions/2.2.3/bin/rake:23:in `load'
/Users/hugobastien/.rbenv/versions/2.2.3/bin/rake:23:in `<top (required)>'
/Users/hugobastien/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/bundler-1.14.6/lib/bundler/cli/exec.rb:74:in `load'
/Users/hugobastien/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/bundler-1.14.6/lib/bundler/cli/exec.rb:74:in `kernel_load'
/Users/hugobastien/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/bundler-1.14.6/lib/bundler/cli/exec.rb:27:in `run'
/Users/hugobastien/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/bundler-1.14.6/lib/bundler/cli.rb:335:in `exec'
/Users/hugobastien/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/bundler-1.14.6/lib/bundler/vendor/thor/lib/thor/command.rb:27:in `run'
/Users/hugobastien/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/bundler-1.14.6/lib/bundler/vendor/thor/lib/thor/invocation.rb:126:in `invoke_command'
/Users/hugobastien/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/bundler-1.14.6/lib/bundler/vendor/thor/lib/thor.rb:359:in `dispatch'
/Users/hugobastien/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/bundler-1.14.6/lib/bundler/cli.rb:20:in `dispatch'
/Users/hugobastien/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/bundler-1.14.6/lib/bundler/vendor/thor/lib/thor/base.rb:440:in `start'
/Users/hugobastien/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/bundler-1.14.6/lib/bundler/cli.rb:11:in `start'
/Users/hugobastien/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/bundler-1.14.6/exe/bundle:32:in `block in <top (required)>'
/Users/hugobastien/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/bundler-1.14.6/lib/bundler/friendly_errors.rb:121:in `with_friendly_errors'
/Users/hugobastien/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/bundler-1.14.6/exe/bundle:24:in `<top (required)>'
/Users/hugobastien/.rbenv/versions/2.2.3/bin/bundle:23:in `load'
/Users/hugobastien/.rbenv/versions/2.2.3/bin/bundle:23:in `<main>'
```

papayaah commented:
yeah - i took care of that on my branch

hugobast commented:
Aight; I'll stop trying to run master how about that 😛

hugobast commented:
Oooops

papayaah commented:
are you testing the `master` branch of this gem? I'm not sure what the state of the master branch is tbh, nor has been ever use...

papayaah commented:
yeah i think v1.4 is the new `master` :D  (v2) is just like an interim branch. I have a PR way back before that would get rid of v2 branch and we can all go back back to v1.4 (it's still open here)

hugobast commented:
Ok glad we settled that; now... I'm running this branch and still have poor luck bundling without issues:

```
Bundler could not find compatible versions for gem "ruby":
  In Gemfile:
    ruby

    solidus was resolved to 1.4.0, which depends on
      ruby (>= 2.1.0)

    solidus_subscriptions was resolved to 1.3.3.beta, which depends on
      ruby (>= 2.3.0)

Could not find gem 'ruby (>= 2.3.0)', which is required by gem 'solidus', in any of the sources.
```

hugobast commented:
Do we give in to the requirements and use ruby 2.3?

papayaah commented:
so odd, i removed the lock file and bundled:


```
➜  solidus_subscriptions git:(fix/syntax-on-subscription-controller) rm Gemfile.lock
➜  solidus_subscriptions git:(fix/syntax-on-subscription-controller) bundle
Using solidus 1.4.0 from git://github.com/solidusio/solidus.git (at v1.4@9290699)
Using solidus_subscriptions 1.3.3.beta from source at `.`
Bundle complete! 27 Gemfile dependencies, 163 gems now installed.
```

also from the gemspec: ```  s.required_ruby_version = '>= 2.3.0' ```

papayaah commented:
so solidus v1.4 does require 2.1:

```https://github.com/solidusio/solidus/blob/v1.4/solidus.gemspec#L14``` 

so odd that it's bundling right away for me.

hugobast commented:
The problem I am seeing is this:

```
▶ cat .ruby-version
2.2.3
```

```
▶ cat solidus_subscriptions.gemspec
# encoding: UTF-8
Gem::Specification.new do |s|
  # ...
  s.required_ruby_version = '>= 2.3.0'
```

hugobast commented:
Yeah I don't know man but `rbenv local 2.3.1` fixed the mismatch

papayaah commented:
i suppose we should downgrade to 2.1 on this gem instead, so we can be compatible with the solidus v1.4. 

odd that Travis CI wasn't complaining either of mismatch ruby version.

hugobast commented:
No no no you misread, the requirements means it has to be above or equal to 2.1 and above or equal to 2.3, that means 2.3+ not 2.1-

hugobast commented:
Ok confirmed; it's working for me locally.

papayaah commented:
oh yea your right, i was thinking about the pessimistic operator

nicklepine commented:
@mickeyren Can we get these tests to go green so we can deploy?

papayaah commented:
@nicklepine there isn't much i can do on the failing Travis. this is only failing in Travis but it's green otherwise.

papayaah commented:
i believe this is 💯 ready to merge

hugobast commented:
I still don't like merging this when it's red...

papayaah commented:
the error seems to be a gateway thing, like Travis trying to connect to our Stripe's test credentials:

```
 1) Editing a subscription payment info can update payment info
     Failure/Error: @credit_card.submit
     CardStore::CardError:
       CardStore::CardError
     # ./app/services/card_store.rb:48:in `gateway_error'
     # ./app/services/card_store.rb:38:in `rescue in create_customer'
     # ./app/services/card_store.rb:35:in `create_customer'
     # ./app/services/card_store.rb:10:in `store_card_for_user'
     # ./app/models/spree/subscription.rb:212:in `block in add_new_credit_card'
```

but seems unable too.

hugobast committed:
Match local version with Travis' version

hugobast committed:
Travis is making http requests to stripe

And this is causing issues with a particular test. In order to avoid
that I reached out to my trusty VCR and threw it at the problematic
spec. I've noticed that other tests are doing HTTP too but I don't want
to do a full pass at this point.
